pkg/cli/cmd/taskrun/describe.go

```diff
@@ -0,0 +1,199 @@
+			if len(args) != 1 {
+				return fmt.Errorf("requires exactly one argument when --uid is not provided")
+			}
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			allNs, _ := cmd.Flags().GetBool("all-namespaces")
+			nsSet := cmd.Flags().Changed("namespace")
+			if allNs && nsSet {
+				return errors.New("cannot use --all-namespaces/-A and --namespace/-n together")
+			}
+			c, err := config.NewConfig(p)
+			if err != nil {
+				return err
+			}
+			opts.Client, err = client.NewRESTClient(c.Get())
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			if len(args) > 0 {
+				opts.ResourceName = args[0]
+			}
+
+			filter := common.BuildFilterString(opts)
+			parent := fmt.Sprintf("%s/results/-", p.Namespace())
+			if opts.AllNamespaces {
+				parent = "*/results/-"
+			}
+
+			recordClient := records.NewClient(opts.Client)
+			resp, err := recordClient.ListRecords(ctx, &pb.ListRecordsRequest{
+				Parent:   parent,
+				Filter:   filter,
+				PageSize: 10,
+			}, "")
+
+			if err != nil {
+				return fmt.Errorf("failed to find TaskRun: %v", err)
+			}
+			if len(resp.Records) == 0 {
+				if opts.UID != "" && opts.ResourceName != "" {
+					return fmt.Errorf("no TaskRun found with name %s and UID %s", opts.ResourceName, opts.UID)
+				} else if opts.UID != "" {
+					return fmt.Errorf("no TaskRun found with UID %s", opts.UID)
+				}
+				return fmt.Errorf("no TaskRun found with name %s", opts.ResourceName)
+			}
+			if len(resp.Records) > 1 {
+				var uids []string
+				for _, record := range resp.Records {
+					uids = append(uids, record.Uid)
+				}
+				return fmt.Errorf("multiple TaskRuns found. Use a more specific name or UID. Available UIDs: %s",
+					strings.Join(uids, ", "))
+			}
+
+			var tr v1.TaskRun
+			if err := json.Unmarshal(resp.Records[0].Data.Value, &tr); err != nil {
+				return fmt.Errorf("failed to unmarshal TaskRun data: %v", err)
+			}
+
+			stream := &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+
+			var data = struct {
+				TaskRun *v1.TaskRun
+				Time    clockwork.Clock
+			}{
+				TaskRun: &tr,
+				Time:    clockwork.NewRealClock(),
+			}
+
+			funcMap := template.FuncMap{
+				"formatAge":       common.FormatAge,
+				"formatDuration":  formatted.Duration,
+				"formatCondition": formatted.Condition,
+				"formatWorkspace": formatted.Workspace,
+				"decorate":        formatted.DecorateAttr,
+			}
+
+			w := tabwriter.NewWriter(stream.Out, 0, 5, 3, ' ', tabwriter.TabIndent)
+			t := template.Must(template.New("Describe TaskRun").Funcs(funcMap).Parse(describeTemplate))
+
+			if err := t.Execute(w, data); err != nil {
+				return err
+			}
+
+			return w.Flush()
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.AllNamespaces, "all-namespaces", "A", false, "use all namespaces")
+	cmd.Flags().StringVar(&opts.UID, "uid", "", "UID of the TaskRun to describe the details")
```

Comment:
```suggestion
	cmd.Flags().StringVar(&opts.UID, "uid", "", "UID of the TaskRun to describe")
```

Reply:
I have done the changes, please take a look